docs-source/src/architecture.rst

```diff
@@ -7,40 +7,70 @@ a prolog integration to Slurm.
 License Manager Agent
 ---------------------
 The ``license-manager-agent`` is responsible for keeping the local cluster license totals
-in sync with the the 3rd party license server totals and also accounting for
-bookings tracked in the backend.
+in sync with the the 3rd party license server totals. It's also responsible for making booking requests
+to the ``license-manager-backend`` when Slurm is configured to use the ``PrologSlurmctld`` script provided by ``license-manager-agent``.
 
-Workload manager support:
+Reconciliation
+**************
+For each license tracked by License Manager, the ``license-manager-agent`` will periodically poll the license server to get
+the usage information and store it in the ``license-manager-backend``. The ``stat-interval`` is the period of time
+between each reconciliation and can be configured in the ``license-manager-agent`` configuration file.
 
-* Slurm
+The information in the ``license-manager-backend`` is used by the reconciliation process to update the license counters in Slurm.
+This is done by creating a reservation to represent the licenses used in the license server.
 
-License server support:
+This reservation is not meant to be consumed by users nor jobs, it's only a representation of the licenses in use.
+The reservation is created by the user configured in the ``license-manager-agent`` configuration file. The user must
+have a user account in the Slurm cluster and have ``operator`` privilege level to manage reservations.
 
-* FlexLM
-* RLM
-* LS-Dyna
-* LM-X
-* OLicense
+Bookings
+********
+The ``license-manager-agent`` is also responsible for making booking requests to the ``license-manager-backend``
+when Slurm is configured to use the ``PrologSlurmctld`` script provided by ``license-manager-agent``.
+
+Each job submitted to Slurm will make a request to the ``license-manager-backend`` to book the needed licenses prior
+to the allocation of the job. The booking ensures that the licenses are available for the job to use, taking into
+consideration the licenses booked for other jobs and the license usage in the license server.
 
+If the booking cannot be made, the job will be kept in the queue until there are enough licenses available to
+satisfy the booking request.
 
-Reconcile
-*********
-The reconcile timed job collects license usage from the 3rd party license servers
-and updates the license-manager-backend on a periodic basis referred to as the ``stat-interval``.
+Grace time
+**********
+Since a job can take some time to check out the license from the license server, each license has a ``grace time``
+period used to indicate when the booking can be deleted, since the job will have already checked out the license
+from the license server and doesn't need to hold the licenses in the cluster anymore.
```

Comment:
These sentences are not super clear, so I suggest a re-wording. I made an attempt here:

```suggestion
A job can take some time to check out the license from the license server after it is submitted to Slurm. Thus, each booking has a ``grace time``
period that is used to indicate how long a booking will be retained. After the grace time expires, the booking is deleted. This ensures that the job has enough time to check out the needed licenses from the license server. A booking is deleted after a job finishes even if there is remaining grace time.
```

Reply:
Thank you for the suggestion, I changed it a little bit to indicate that the grace time is related to the license, not to the booking. Does it sound better now?